pkg/plugin/oauth2/setup.go

```diff
@@ -1,14 +1,27 @@
 package oauth2
 
 import (
+	"net/url"
+
+	"github.com/hellofresh/janus/pkg/config"
 	"github.com/hellofresh/janus/pkg/jwt"
-	"github.com/hellofresh/janus/pkg/oauth"
+	"github.com/hellofresh/janus/pkg/notifier"
 	"github.com/hellofresh/janus/pkg/plugin"
 	"github.com/hellofresh/janus/pkg/proxy"
+	"github.com/hellofresh/janus/pkg/router"
+	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 )
 
+var (
+	repo        Repository
+	adminRouter router.Router
+)
+
 func init() {
+	plugin.RegisterEventHook(plugin.StartupEvent, onStartup)
```

Comment:
That's for getting rid of the globals variables declared on the top, but perhaps we could keep an eyes on it, leave the way it is, and see if it really make sense to design the way I suggested